controllers/LikesController.js

```diff
@@ -0,0 +1,166 @@
+import models from '../database/models';
+import ServerResponse from '../modules';
+
+
+const { notFoundError, successResponse } = ServerResponse;
+
+/**
+ * @class likeController
+ * @exports likeController
+ */
+export default class likeController {
+  /**
+   *
+   * @method likePost
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @param {function} next
+   * @returns {object} returns...
+   */
+  static async likeArticle(req, res, next) {
+    try {
+      const { slug } = req.params;
+      const userId = req.user.id;
+
+      const article = await models.Article.findOne({ where: { slug } });
+      if (!article) return notFoundError(req, res);
+
+      const articleId = article.dataValues.id;
+
+      const previousChoice = await models.Like.findOne({ where: { userId, articleId } });
+
+      // Check for previous like on article and undo
+      if (previousChoice && (previousChoice.like === true)) {
+        await models.Like.destroy({ where: { userId, articleId } });
+
+        // Update count on article table
+        await models.Article.update(
+          { likes: (article.dataValues.likes - 1) },
+          { where: { slug } },
+        );
+
+        return successResponse(
+          res, 200, 'message',
+          `Your like on '${article.dataValues.title}' has been removed`
+        );
+      }
+      // Check if user previously disliked article (change dislike to like if true)
+      if (previousChoice && (previousChoice.like === false)) {
+        await models.Like.update(
+          { like: true },
+          { where: { userId, articleId } },
+        );
+
+        // Update count on article table
+        await models.Article.update(
+          {
+            likes: (article.dataValues.likes + 1),
+            dislikes: (article.dataValues.dislikes - 1),
+          },
+          { where: { slug } },
+        );
+        return successResponse(
+          res, 200, 'message',
+          `You just liked '${article.dataValues.title}'`,
+        );
+      }
+
+      // Create new like if all above checks fail
+      await models.Like.create({
+        articleId,
+        userId,
+        like: true,
+      });
+
+      // Update count on article table
+      await models.Article.update(
+        { likes: (article.dataValues.likes + 1) },
+        { where: { slug } },
+      );
+
+      return successResponse(
+        res, 200, 'message',
```

Comment:
Shouldn't this be a `201`?

Reply:
Yes it should. I'll change it right away.